dialects/regex/pass/dfa2matcher.cpp

```diff
@@ -0,0 +1,173 @@
+#include "dfa2matcher.h"
+
+#include <algorithm>
+#include <sstream>
+
+#include "thorin/def.h"
+
+#include "dialects/core/core.h"
+#include "dialects/mem/mem.h"
+#include "dialects/regex/pass/dfa.h"
+#include "dialects/regex/range_helper.h"
+
+using namespace thorin;
+
+// nomenclature:
+// c is the character from the string we want to match
+
+// see lit/regex/match_manual.thorin for a hand written state machine impl that this is based on
+
+// the main idea is:
+// for every state in the DFA, we create a lam that checks if the char c at pos is the end of the string \0
+// if not, jump to a checker lam, that consists of a few bit-wise or-ed in-range checks to verify if we can transition
+// to a certain other state with c if any of the checks is true, we jump to the corresponding state lam with the updated
+// position if the check fails, we jump to the next checker lam or the exit lam if we checked all possible transitions
+// Note, since the DFA stores the transitions as chars, not ranges, we have to merge the transitions to ranges first
+// (cf. transitions_to_ranges)
+
+#define DEBUG_PRINT 0
```

Comment:
use `world.DLOG(...)` instead

Reply:
Hmm... I'm a bit hesitant to "clutter" the pure automaton impl. stuff with thorin implementation stuff, i.e. passing through the world just to be able to debug print feels a bit off.. but dunno...

Reply:
I just grep'ed through the code. There are three instances of `#if DEBUG_PRINT` - each of those has the `World` available anyway...